app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -96,58 +96,78 @@ class WallpapersUseCases(
             // and download utilities.
             withContext(Dispatchers.IO) {
-                val availableWallpapers = getAvailableWallpapers()
+                val availableWallpapers = possibleWallpapers.getAvailableWallpapers()
                 val currentWallpaperName = settings.currentWallpaper
                 val currentWallpaper = possibleWallpapers.find { it.name == currentWallpaperName }
                     ?: fileManager.lookupExpiredWallpaper(currentWallpaperName)
                     ?: Wallpaper.Default
 
                 fileManager.clean(
                     currentWallpaper,
-                    possibleWallpapers.filterIsInstance<Wallpaper.Remote>()
+                    possibleWallpapers
                 )
                 downloadAllRemoteWallpapers(availableWallpapers)
                 store.dispatch(AppAction.WallpaperAction.UpdateAvailableWallpapers(availableWallpapers))
                 store.dispatch(AppAction.WallpaperAction.UpdateCurrentWallpaper(currentWallpaper))
             }
         }
 
-        private fun getAvailableWallpapers() = possibleWallpapers
-            .filter { !it.isExpired() && it.isAvailableInLocale() }
+        private fun List<Wallpaper>.getAvailableWallpapers() =
+            this.filter { !it.isExpired() && it.isAvailableInLocale() }
 
         private suspend fun downloadAllRemoteWallpapers(availableWallpapers: List<Wallpaper>) {
-            for (wallpaper in availableWallpapers.filterIsInstance<Wallpaper.Remote>()) {
+            for (wallpaper in availableWallpapers) {
+                if (wallpaper == Wallpaper.Default) continue
                 downloader.downloadWallpaper(wallpaper)
             }
         }
 
-        private fun Wallpaper.isExpired(): Boolean = when (this) {
-            is Wallpaper.Remote -> {
-                val expired = this.expirationDate?.let { Date().after(it) } ?: false
-                expired && this.name != settings.currentWallpaper
-            }
-            else -> false
+        private fun Wallpaper.isExpired(): Boolean {
+            val expired = this.endDate?.let { Date().after(it) } ?: false
+            return expired && this.name != settings.currentWallpaper
         }
 
         private fun Wallpaper.isAvailableInLocale(): Boolean =
-            if (this is Wallpaper.Promotional) {
-                this.isAvailableInLocale(currentLocale)
-            } else {
-                true
-            }
+            this.availableLocales?.contains(currentLocale) ?: true
 
         companion object {
-            private val localWallpapers: List<Wallpaper.Local> = listOf(
-                Wallpaper.Local.Firefox("amethyst", R.drawable.amethyst),
-                Wallpaper.Local.Firefox("cerulean", R.drawable.cerulean),
-                Wallpaper.Local.Firefox("sunrise", R.drawable.sunrise),
+            private val localWallpapers: List<Wallpaper> = listOf(
+                Wallpaper(
+                    "amethyst",
```

Comment:
nit: missed the named parameter for `name` for amethyst and cerulean and beach-vibe

Reply:
ty!